lib/matplotlib/backends/backend_webagg_core.py

```diff
@@ -360,6 +278,31 @@ def _handle_set_device_pixel_ratio(self, device_pixel_ratio):
             self._force_full = True
             self.draw_idle()
 
+    def handle_save(self, event):
+        figure_id = event['figure_id']
+        format = event['format']
+
+        mimetype = mimetypes.types_map.get(f".{format}")
+        if mimetype is None:
+            window.alert(f"Cannot download plot, unable to determine mimetype for '{format}'")
+            return
+
+        element = document.createElement('a')
+        data = BytesIO()
+        self.figure.savefig(data, format=format)
+
+        element.setAttribute(
+            "href",
+            "data:{};base64,{}".format(
+                mimetype, base64.b64encode(data.getvalue()).decode("ascii")
+            ),
+        )
+        element.setAttribute("download", f"plot{figure_id}.{format}")
+        element.style.display = "none"
+        document.body.appendChild(element)
+        element.click()
+        document.body.removeChild(element)
```

Comment:
I would have sworn there was a better way to do this by now. But maybe not?

Reply:
I haven't found a better way of doing this yet, so I am leaving it as it is for the moment.